staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -1296,6 +1352,23 @@ func retryAfterSeconds(resp *http.Response) (int, bool) {
 	return 0, false
 }
 
+// Context uses method chaining function to set the global context of the request.
+// It's best to be the first in the chain, like:
+//
+//	Request{}.Context(ctx).Namespace(ns).Name(n).Resource(res).Do()
+func (r *Request) Context(ctx context.Context) *Request {
+	r.gctx = ctx
+	return r
+}
+
+// ctx returns the global context, if it is not set, we use the default context.Background()
```

Comment:
Presumably this is needed because `klog.FromContext` and `logr.FromContext` crash when passed a nil context. Probably too late to change that now.

`getCtx` returning `r.ctx`?

Reply:
Yes, the function returns the member field `r.ctx`, but if the member field is nil, the default value `context.Background()` is returned.